src/lib/codec-wrappers/codec.ts

```diff
@@ -0,0 +1,7 @@
+export interface Encoder {
+  encode(data: ImageData): Promise<ArrayBuffer | SharedArrayBuffer>;
```

Comment:
What causes the return type to be `ArrayBuffer` *or* `SharedArrayBuffer`?

Reply:
Also, if this going to be used by anything? Won't they all have different options?

Reply:
I’ll re-check `ArrayBuffer` vs `SharedArrayBuffer`, but I think TS was complaining about _something_ (and was, strictly speaking, correct).

Reply:
> I did this because I thought it would be nice to have a consistent encode() call and rather have a stateful encoder with individual calls to set options.

Yeah, fair enough!

Reply:
Yeah, `SharedArrayBuffer` was a left-over. I just removed it :)